### PR TITLE
fix(es/minifier): use seperate scope for class

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**
This would fix two issues --
```js
export class A {
   foo() {
      var o = this
      return class {
        x = console.log(o)  
      }
   }
}
```
and
```js
export var a

(a = new Foo())
```

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc/tests/vercel/full/next-33088/output/index.js [hunk 1]
```diff
@@ -14,8 +14,8 @@ export default function i() {
         function t() {
             return (t = r(function() {
                 var r, t;
-                return e(this, function(e) {
-                    switch(e.label){
+                return e(this, function(r) {
```
Comment (kdy1) [inline crates/swc/tests/vercel/full/next-33088/output/index.js:17]: This seems wrong.

File: crates/swc/tests/vercel/full/next-33088/output/index.js [hunk 2]
```diff
@@ -24,10 +24,10 @@ export default function i() {
                         case 1:
                             return [
                                 4,
-                                (r = e.sent()).json()
+                                r.sent().json()
```
Comment (kdy1) [inline crates/swc/tests/vercel/full/next-33088/output/index.js:27]: `r = e` is only true if this line is evaluated

File: crates/swc/tests/vercel/full/utf8-1/output/index.js [hunk 3]
```diff
@@ -29,17 +29,17 @@ function p() {
                 case 2:
                     return [
                         4,
-                        (e = t.sent()).json()
+                        e.sent().json()
```
Comment (kdy1) [inline crates/swc/tests/vercel/full/utf8-1/output/index.js:32]: Same here

**Review by kdy1 — COMMENTED** [review]
Thanks!

File: crates/swc/tests/vercel/full/next-33088/output/index.js (hunk 2 above)
Comment (Austaras) [reply crates/swc/tests/vercel/full/next-33088/output/index.js:27]: r and e doesn't get referenced anywhere, so assignment is removed. You can repplicate this in terser, swc currently only needs more passes to remove unused vars.

Comment (kdy1) [reply crates/swc/tests/vercel/full/next-33088/output/index.js:27]: I see. My bad, thank you!

**Review by kdy1 — APPROVED** [review]
Thank you!
I'll merge https://github.com/swc-project/swc/pull/5772 after merging this.

---

swc-bump:
 - swc_ecma_minifier

**Pull request merged**